app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt

```diff
@@ -206,6 +208,26 @@ class MainActivity : BaseActivity() {
     }
 
     private fun setupBottomNavigation() = with(binding) {
+        bottomNavigation.findViewById<View>(R.id.menuFragment).setOnLongClickListener {
+            navController.navigate(R.id.switchUserActivity)
+            true
+        }
+
+        bottomNavigation.findViewById<View>(R.id.menuFragment).setOnClickListener {
+            navController.navigate(R.id.menuFragment)
+
+            if (it.tag != null && Date().time - (it.tag as Date).time < DOUBLE_TAP_DELAY) {
+                CoroutineScope(Dispatchers.IO).launch {
+                    trackAccountEvent("switch")
+                    AccountUtils.switchToNextUser()
+                }.also { navController.navigate(R.id.homeFragment) }
+
+                it.tag = null
+            } else {
+                it.tag = Date()
+            }
```

Comment:
Instead of bothering with manually computing the time to detect double tap we can use the listeners that are made for this very purpose:

```kt
val gestureDetector = GestureDetector(this@MainActivity, object : SimpleOnGestureListener() {
    override fun onDoubleTap(e: MotionEvent): Boolean {
        // TODO: Switch user
        return true
    }
})

profileItem.setOnTouchListener { v, event -> gestureDetector.onTouchEvent(event) }
```

Reply:
I came across this when searching for a solution, but could not make it to work. In fact, I was missing the last line. It is now implemented, thank you!